### PR TITLE
refactor: replace reifyEff with typematch in stdlib

### PR DESCRIPTION
**Pull request by mlutze**
related to #4773

### CODE REVIEW

**Comment by magnus-madsen** [comment]
Just a double check to ensure no silly mistakes :)

File: main/src/library/DelayMap.flix [hunk 1]
```diff
@@ -154,17 +154,17 @@ namespace DelayMap {
     ///
     @Experimental @ParallelWhenPure
     pub def count(f: (k, v) -> Bool \ ef, m: DelayMap[k, v]): Int32 \ ef =
-        let f1 = match (k, v) -> f(k, v);
         def c() = foldLeftWithKey((b, k, v) -> if (f(k, v)) b + 1 else b, 0, m);
-        reifyEff(f1) {
-            case Pure(g) =>
+        typematch f {
+            case g: (k, v) -> Bool \ {} =>
```
Comment (magnus-madsen) [inline main/src/library/DelayMap.flix:159]: Does k, v here refer to the params in the signature k, v? I assume so, right?

Comment (magnus-madsen) [reply main/src/library/DelayMap.flix:159]: Why do you write `\ {}` here when you did not in `DelayList`? 

I guess it makes sense to be explicit and write `\ {}` everywhere?

Comment (mlutze) [reply main/src/library/DelayMap.flix:159]: > Does k, v here refer to the params in the signature k, v? I assume so, right?

Yes. No new variables can be introduced in a typematch (except wildcards `_`).



> Why do you write `\ {}` here when you did not in `DelayList`?
> 
> I guess it makes sense to be explicit and write `\ {}` everywhere?

It also nice lines up with `\ ef`.

Comment (magnus-madsen) [reply main/src/library/DelayMap.flix:159]: I suggest to write `\ {}` everywhere then! :)

**Review by jaschdoc — APPROVED** [review]
Looks good. I agree with Magnus that there should be explicit type casts to `\ {}` :)

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**